src/externalized/ItemRange.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <vector>
+#include "ItemModel.h"
+
+class ItemRange {
+	using Iterator = std::vector<ItemModel const *>::const_iterator;
+
+	public:
+		ItemRange(Iterator begin, Iterator end) : m_begin(begin), m_end(end) {};
+		Iterator begin() {
+			return m_begin;
+		};
+		Iterator end() {
+			return m_end;
+		};
+
+	private:
+		Iterator m_begin;
+		Iterator m_end;
+};
```

Comment:
The methods of this class could all be marked `const noexcept`  which might help the compiler optimize them.

Reply:
Fixed.